src/main/java/com/linecorp/armeria/common/MimeTypeParams.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+final class MimeTypeParams {
+
+    static String find(String params, String name) {
```

Comment:
I think you should be able to use a regex to simplify this without losing performance, does something like this work?

```java
private static final Pattern PARAM_FINDER = Pattern.compile(";\s*([^=\s])+\s*=\s*([^;\s]+)");

Matcher m = PARAM_FINDER.matcher(params);
while(m.find(params)) {
  if (m.group(1).equals(name)) {
    return maybeUnquote(m.group(2));
  }
}
return null;
```